docs/developers/debugging.md

```diff
@@ -0,0 +1,133 @@
+
+
+## Environment
+
+Users can supply CLI arguments to their node indirectly by populating the relevant environment variables. The list of CLI arguments and their environment variables can be found by running the `help` command.
+
+Users can define environment variables by configuring their shell or setting them inside the current terminal.
+
+Additionally, users can add a `.env` file to the root directory of their local copy of their Fuel node (or wherever the working directory is). Using a `.env` file requires compiling the client with the `env` feature. This can be accomplished by including `env` in the list of features or by passing the `--all-features` flag. 
+
+Using environment variables for CLI arguments allows these values to be reused across IDEs and IDE configurations seamlessly.
+
+
+## P2P Network
+
+When running the client with the P2P service enabled, i.e., building the binary with the `p2p` feature and supplying the runtime argument `--enable-p2p`, the client will connect to a Fuel network. This requires additional CLI arguments, including the `--keypair` and `--network` arguments.
+
+A key pair can be generated by running the utility binary `fuel-core-keygen`:
+
+```bash
+cargo run --bin fuel-core-keygen new
+```
+
+The resulting key pair printed to the console contains an address and secret. The `secret` can be provided to the `--keypair` argument when running the node.
+
+The `--network` argument identifies the name of the network to join. The network name is used during peer discovery. For example, users can specify `--network beta-4` to join the Beta 4 network. Similarly, setting the environment variable `NETWORK="beta-4"` will produce the same result.
+
+For more information about client networking, see the Fuel guide on [running a node](https://docs-hub.vercel.app/guides/running-a-node/).
+
+## Recommended IDEs
+
+During development, it is recommended to delegate the execution of Cargo instructions to an IDE and connect the application to the IDE's debugger, instead of running these instructions manually.
+
+Recommended IDEs include:
+- CLion
+- Visual Studio Code
+
+## CLion
+
+### Prerequisites
+
+Running Fuel Core with CLion requires the official JetBrains Rust plugin. Find the Rust plugin by navigating to CLion's Plugin menu, entering "Rust" in the search. Install it by clicking "Install".
+
+### Create A Run Configuration
+
+1. In the top-right corner of the CLion window, click on the configurations dropdown menu and select "Edit Configurations..."
+2. In the "Run/Debug Configurations" window, click on the "+" button and select "Cargo"
+3. In the "Name" field, give your configuration a descriptive name, such as "Run Beta 3"
```

Comment:
```suggestion
3. In the "Name" field, give your configuration a descriptive name, such as "Run Fuel Client"
```

Just to remove the need to update the docs. Maybe is better to set a generic name

Reply:
Yep, that makes sense 👍